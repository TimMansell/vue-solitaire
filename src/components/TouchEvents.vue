<template>
  <div ref="swipe">
    <slot />
  </div>
</template>

<script>
import Hammer from 'hammerjs';

export default {
  name: 'TouchEvents',
  props: {
    disabled: {
      type: Boolean,
      default: true,
    },
  },
  mounted() {
    this.init();
  },
  methods: {
    init() {
      const hammertime = new Hammer(this.$refs.swipe);

      hammertime.get('swipe').set({ direction: Hammer.DIRECTION_VERTICAL });

      hammertime.on('swipe', () => {
        this.swipe();
      });

      hammertime.on('doubletap', () => {
        this.doubleTap();
      });
    },
    swipe() {
      if (this.disabled) return;

      this.$emit('swipe');
    },
    doubleTap() {
      if (this.disabled) return;

      this.$emit('doubletap');
    },
  },
};
</script>
