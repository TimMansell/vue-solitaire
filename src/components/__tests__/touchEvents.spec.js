import { shallowMount } from '@vue/test-utils';
import TouchEvents from '@/components/TouchEvents.vue';

describe('TouchEvents.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(TouchEvents);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call swipe method', () => {
    const wrapper = shallowMount(TouchEvents, {
      propsData: {
        disabled: false,
      },
    });

    wrapper.vm.swipe();

    expect(wrapper.emitted().swipe).toBeTruthy();
  });

  it('should call doubleTap method', () => {
    const wrapper = shallowMount(TouchEvents, {
      propsData: {
        disabled: false,
      },
    });

    wrapper.vm.doubleTap();

    expect(wrapper.emitted().doubletap).toBeTruthy();
  });
});
