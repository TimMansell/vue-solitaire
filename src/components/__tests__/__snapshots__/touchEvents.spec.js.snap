// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchEvents.vue matches snapshot 1`] = `<div style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);"></div>`;
